From 39d21409e444b47984a8235779ee2948999471d8 Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Tue, 12 May 2020 07:27:27 +0000
Subject: [PATCH] vrf: change the search pattern for os-release

The contents of /etc/os-release changes a
little after below commit introduced in
oe-core:
f451c68667 os-release: sanitize required fields

Before the commit:
 # cat /etc/os-release
ID="wrlinux-cgl"
NAME="Wind River Linux CGL"
VERSION="10.20.20.0"
VERSION_ID="10.20.20.0"
PRETTY_NAME="Wind River Linux CGL 20.20"

After the commit:
 # cat /etc/os-release
ID=wrlinux-cgl
NAME="Wind River Linux CGL"
VERSION="10.20.20.0"
VERSION_ID=10.20.20.0
PRETTY_NAME="Wind River Linux CGL 20.20"

So we need to adjust the search pattern
for ID and VERSION_ID line to gurantee to
construct the expected VRF_DISTRIB_ID and
VRF_DISTRIB_RELEASE_VER respectively.

Otherwise it will use inexistent /vrf/etc/package/__package.list
instead of the /vrf/etc/package/wrlinux-cgl_10.20.20.0_package.list
(take the version 10.20.20.0 for example)
which used to define a lot of variables used
to define how to install vrf.

Upstream-Status: Inappropriate [no upstream]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 src/vrf-helper.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/vrf-helper.in b/src/vrf-helper.in
index ec88b96..129314c 100755
--- a/src/vrf-helper.in
+++ b/src/vrf-helper.in
@@ -379,8 +379,8 @@ initialize_env () {
         VRF_DISTRIB_ID=$(grep "DISTRIB_ID=" /etc/lsb-release| \
                                   sed -e "s/DISTRIB_ID=//")
     elif [ -e /etc/os-release ]; then
-        VRF_DISTRIB_ID=$(sed -n 's/^ID="\(.*\)"/\1/p' /etc/os-release)
-        VRF_DISTRIB_RELEASE_VER=$(sed -n 's/^VERSION_ID="\(.*\)"/\1/p' /etc/os-release)
+        VRF_DISTRIB_ID=$(sed -n 's/^ID=\(.*\)/\1/p' /etc/os-release)
+        VRF_DISTRIB_RELEASE_VER=$(sed -n 's/^VERSION_ID=\(.*\)/\1/p' /etc/os-release)
     elif [ -z "$VRF_DISTRIB_RELEASE_VER" -o -z "$VRF_DISTRIB_ID" ]; then
         echo "No information about release, please set env"
         echo "VRF_DISTRIB_RELEASE_VER and VRF_DISTRIB_ID"
-- 
2.24.1

